### DISCUSSION TITLE
Installing the Library

### DISCUSSION THREAD

**Discussion by IssaAboudi** [nlohmann/json#3214] [Q&A] [answered]
Hello friends:

I'm really new to installing libraries and adding them to my project and I was curious as to how I could do that in CMake with this library. I recognize I am missing a lot of prerequisite knowledge and I would appreciate any help

This is the error I am getting:
![image](https://user-images.githubusercontent.com/65807151/147375094-0f4d0b86-dd3b-4ec8-b3f5-50bbca1b7b40.png)

my CmakeLists.txt looks like:
![image](https://user-images.githubusercontent.com/65807151/147375102-53360f6c-b229-4106-8a3a-a50ad16a665e.png)

**Comment by nlohmann** [comment, marked as answer]
Did you check https://json.nlohmann.me/integration/cmake/

**Reply by IssaAboudi** [reply]
I've looked at it but I don't think I understand CMake enough to be able to import this using the instructions found there. It's 100% a me not understanding how to do this kind of problem.

I have two questions that might get me closer to better understanding the instructions:
1) What exactly am I importing from the github into my project, just the folder in include? Or the entire github? 
2) I notice there is a folder called CMake. Do i need that in my project?

**Reply by nlohmann** [reply]
If you want to use the library with CMake, it is easiest to import the complete project. This is what is described in https://json.nlohmann.me/integration/cmake/#embedded (where the content of the project folder is assumed to be `nlohmann_json`:

```cmake
add_subdirectory(nlohmann_json)
```

**Reply by nlohmann** [reply]
@IssaAboudi Do you need further assistance with this issue?

**Reply by IssaAboudi** [reply]
Hey no I'm good thanks
